src/Entity/DatasetLink.php

```diff
@@ -0,0 +1,264 @@
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text", nullable=true)
+     */
+    protected $url;
+
+    /**
+     * Name for the Dataset Link.
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text", nullable=true)
+     */
+    protected $name;
+
+    /**
+     * Description for the Dataset Link.
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text", nullable=true)
+     */
+    protected $description;
+
+    /**
+     * Function code for the Dataset Link.
+     *
+     * @var string
+     *
+     * @see ONLINE_FUNCTION_CODES
+     *
+     * @ORM\Column(type="text", nullable=true)
+     */
+    protected $functionCode;
+
+    /**
+     * Protocol for the Dataset Link.
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text", nullable=true)
+     */
+    protected $protocol;
+
+    /**
+     * Protocol for the Dataset Link.
+     *
+     * @var string
+     *
+     * @ORM\ManyToOne(targetEntity="App\Entity\DatasetSubmission", inversedBy="datasetLinks")
+     * @ORM\JoinColumn(nullable=false)
+     */
+    private $datasetSubmission;
+
+    /**
+     * Get the URL for the Dataset Link.
+     *
+     * @return string
+     */
+    public function getUrl(): ?string
+    {
+        return $this->url;
+    }
+
+    /**
+     * Set the URL for this Dataset Link.
+     *
+     * @param string $url The URL for this Dataset Link.
+     *
+     * @return self
+     */
+    public function setUrl(?string $url): self
+    {
+        $this->url = $url;
+
+        return $this;
+    }
+
+    /**
+     * Get the name for the Dataset Link.
+     *
+     * @return string
+     */
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    /**
+     * Set the name for this Dataset Link.
+     *
+     * @param string $name The name for this Dataset Link.
+     *
+     * @return self
+     */
+    public function setName(?string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Get the description for the Dataset Link.
+     *
+     * @return string
+     */
+    public function getDescription(): ?string
+    {
+        return $this->description;
+    }
+
+    /**
+     * Set the description for the Dataset Link.
+     *
+     * @param string $description The description for the Dataset Link.
+     *
+     * @return self
+     */
+    public function setDescription(?string $description): self
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+
+    /**
+     * Get the function code for the Dataset Link.
+     *
+     * @return string
+     */
+    public function getfunctionCode(): ?string
+    {
+        return $this->functionCode;
+    }
+
+    /**
+     * Set the function code for the Dataset Link.
+     *
+     * @param string $functionCode The function code for the Dataset Link.
+     *
+     * @return self
+     */
+    public function setFunctionCode(?string $functionCode): self
```

Comment:
Same here

Reply:
I didn't see your prior comment referred to, but would suggest checking that the past code is in the list of constants above.